app/assets/javascripts/editor.js

```diff
@@ -27,93 +29,112 @@ class Editor {
       }
     });
   }
-  setState(commentFormID = "main") {
+  setState(commentFormID) {
     this.commentFormID = commentFormID;
-    $E.textarea = $("#text-input-" + commentFormID);
+    this.textarea = $("#text-input-" + commentFormID);
     $E.textarea.bind('input propertychange', $E.save);
-    $E.preview = $("#comment-preview-" + commentFormID);
+  }
+  get textAreaElement() {
+    const textAreaID = "#text-input-" + this.commentFormID;
+    return $(textAreaID);
+  }
+  get textAreaValue() { 
+    return this.textAreaElement.val(); 
+  }
+  get previewElement() {
+    let previewIDPrefix = "#comment-preview-";
+    if (this.elementIDPrefixes.hasOwnProperty("preview")) {
+      // eg. on /wiki/new & /wiki/edit, the preview element is called #preview-main
+      previewIDPrefix = "#" + this.elementIDPrefixes["preview"] + "-";
```

Comment:
This is one of the main changes in this PR. Honestly, I have mixed feelings about it since it's a little weird, especially in the constructor.

But I do think it's an okay workaround. I made this because there was some tension in having `editor.js` operative on both 1) multi-comment pages and 2) `/wiki/new` and `/wiki/edit`. 

I wanted preview elements to be named `#comment-preview-reply-123` (they used to just be called `#preview`), for greater transparency. But that clashes with the wiki editor, where there are no comments involved.

So, this is my workaround. It's a little weird, but thanks to OOP, it can be scaled back or revised much easier.